src/fastspell/ai.py

```diff
@@ -0,0 +1,200 @@
+
+        # print(prediction)
+
+        lang0 = prediction[0][0][len(ft_prefix):]
+        lang0_prob = prediction[1][0]
+        if len(prediction[0]) >= 2:
+            lang1 = prediction[0][1][len(ft_prefix):]
+            lang1_prob = prediction[1][1]
+        else:
+            # If there's only one option... Not much to do.
+            continue
+        if len(prediction[0]) >= 3:
+            lang2 = prediction[0][2][len(ft_prefix):]
+            lang2_prob = prediction[1][2]
+        else:
+            lang2 = None
+            lang2_prob = 0.0
+
+        label = None
+        if lang0 == lang.alpha_2:
+            label = 0
+        elif lang1 == lang.alpha_2:
+            label = 1
+        elif lang2 == lang.alpha_2:
+            label = 2
+
+        if label is None:
+            continue
+
+        # print(lang0)
+
+        raw_tokens = text.strip().split(" ")
+        if lang0 in hunspell_objs:
+            tokens = remove_unwanted_words(raw_tokens, lang0)
+            correct = 0
+            for token in tokens:
+                try:
+                    if hunspell_objs[lang0].spell(token):
+                        correct += 1
+                except UnicodeEncodeError as ex:
+                    pass
+            lang0_dic_tokens = correct / len(tokens)
+        else:
+            lang0_dic_tokens = None
+
+        if lang1 in hunspell_objs:
+            tokens = remove_unwanted_words(raw_tokens, lang1)
+            correct = 0
+            for token in tokens:
+                try:
+                    if hunspell_objs[lang1].spell(token):
+                        correct += 1
+                except UnicodeEncodeError as ex:
+                    pass
+            lang1_dic_tokens = correct / len(tokens)
+        else:
+            lang1_dic_tokens = None
+
+        if lang2 in hunspell_objs:
+            tokens = remove_unwanted_words(raw_tokens, lang2)
+            
+            correct = 0
+            for token in tokens:
+                try:
+                    if hunspell_objs[lang2].spell(token):
+                        correct += 1
+                except UnicodeEncodeError as ex:
+                    pass
+
+            lang2_dic_tokens = correct / len(tokens)
+        else:
+            lang2_dic_tokens = None
+
+        sentences.append({
+            "fastText_lang0": lang0_prob,
+            "fastText_lang1": lang1_prob,
+            "fastText_lang2": lang2_prob,
+            "lang0_dic_tokens": lang0_dic_tokens,
+            "lang1_dic_tokens": lang1_dic_tokens,
+            "lang2_dic_tokens": lang2_dic_tokens,
+        })
+        labels.append(label)
+
+        # count += 1
+        # if count == 7:
+        #     break
+
+print(len(sentences))
+
+dict_vectorizer = DictVectorizer()
+X = dict_vectorizer.fit_transform(sentences)
+
+xgb_model = xgboost.XGBClassifier(n_jobs=10)
+
+clf = GridSearchCV(
+    xgb_model,
+    {"max_depth": [1, 2, 4, 6], "n_estimators": [25, 50, 100, 200]},
```

Comment:
We could presumably introduce another hyperparameter, a "threshold" above which we don't need to go look at the dictionary (e.g. if fastText is 99% sure it's Italian, we don't need to check further). This could lower the cost of the approach.